### PR TITLE
Wk 8 Synthesizing in progress

### PR DESCRIPTION
**Pull request by Logan213**
Chad - can you look at my code in progress?  I have gotten the 'while loop' to trigger and prompt for a password, but if I put in the correct one (or not), it just loops back to prompting for the password again. Basically, it's skipping my "if" statement and I can't figure out why. Any advice?

### CODE REVIEW

File: task_02.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        >>>
+
+    """
+    authenticated = False
+
+    while not authenticated:
+        passinput = getpass.getpass()
+        print passinput
+        cred = authentication.authenticate(username, passinput)
+        if cred == 'Yes':
```
Comment (cheuschober) [inline task_02.py]: @Logan213 Check the output of `authentication.authenticate()` - It's not returning a string.

Comment (Logan213) [inline task_02.py]: @cheuschober Thanks for the tip! For whatever reason I swore it was returning 'Yes' or 'No'.  Safe travels!

File: task_02.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    while not authenticated and maxattempts != 0:
+        passinput = getpass.getpass()
+        print passinput
+        authenticated = authentication.authenticate(username, passinput)
+        if authenticated is True:
+            return authenticated
+        elif not authenticated and maxattempts != 0:
+            maxattempts -= 1
+            print attleft.format(maxattempts)
+    return authenticated
```
Comment (cheuschober) [inline task_02.py]: @Logan213 Glad to hear it helped. FYI, there are a few violations of OAOO and DRYness in here that you might want to rectify. Nothing that would cost you this round but could affect the midterm if I saw similar. Notably, you have a lot more tests (`if` and `elif`) than you need (and an unnecessary `print` statement). You can rely on the `while` condition to do its job.

File: task_01.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        >>> get_matches(['Chad', 'Modesto'])
+        [('Chad', 'Modesto')]
+
+    """
+    mylist = []
+    for idx1, name in enumerate(players):
+        for idx2, name in enumerate(players):
+            if idx1 < idx2:
+                mylist.append((players[idx1], name))
+    return mylist
```
Comment (cheuschober) [inline task_01.py]: @Logan213, I think you can do this without `players[idx1]` which costs you a lookup. Note how you named each round of the indexes differently as `idx1` and `idx2` -- what's keeping you from doing the same with the names?

**Pull request merged**